### PR TITLE
fix: Disallow attendee editing if order status not initializing

### LINKED ISSUE #7066: Disallow attendee editing after order is in pending+ state
Right now, there is no check on attendee editing being disallowed based on order state, this should be fixed

### PR DESCRIPTION
**Pull request by Haider8**
Fixes #7066 

#### Short description of what this resolves:
Attendee details can be edited even if the status of the corresponding order is not `initializing`.

#### Changes proposed in this pull request:
Attendee can no longer be edited if the order status is not `initializing`.

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/attendees.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5767154&bid=18875279)

File: app/api/attendees.py [hunk 1]
```diff
@@ -206,6 +206,12 @@ def before_update_object(self, obj, data, kwargs):
         :param kwargs:
         :return:
         """
+        order = safe_query(Order, 'id', obj.order_id, 'id')
```
Comment (iamareebjamal) [inline app/api/attendees.py:209]: Use new method, safe_query_by_id

File: app/api/attendees.py [hunk 2]
```diff
@@ -206,6 +206,12 @@ def before_update_object(self, obj, data, kwargs):
...
+        if order.status != 'initializing':
+            raise UnprocessableEntityError(
+                {'pointer': '/data/id'}, "Order status is not initializing",
```
Comment (iamareebjamal) [inline app/api/attendees.py:212]: This is not proper error. You should say that attendee can't be updated and why

**Comment by iamareebjamal** [comment]
Add a test for the appropriate cases, like places, completed, pending.

Also, attendee should not be able to be deleted as well once it is created

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/attendees.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5767154&bid=18875279)

**Comment by Haider8** [comment]
> Also, attendee should not be able to be deleted as well once it is created

@iamareebjamal Yes, right now we can't delete attendee if order status is initializing, but the error message is same as that of edit attendee.

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/attendees.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5767154&bid=18875279)

**Comment by iamareebjamal** [comment]
OK. Please fix dredd tests and add new tests

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/attendees.py  1
         

Clones added
============
- tests/all/integration/api/attendee/test_attendee_api.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5767154&bid=18875279)

File: tests/all/integration/api/attendee/test_attendee_api.py [hunk 3]
```diff
@@ -520,3 +520,75 @@ def test_edit_attendee_order(db, client, jwt):
...
+
+    db.session.commit()
+
+    attendee_order = attendee.order
+
+    data = json.dumps(
+        {
+            'data': {
+                'type': 'attendee',
+                'id': str(attendee.id),
```
Comment (iamareebjamal) [inline tests/all/integration/api/attendee/test_attendee_api.py:575]: You should modify some attributes here and

File: tests/all/integration/api/attendee/test_attendee_api.py [hunk 4]
```diff
@@ -520,3 +520,75 @@ def test_edit_attendee_order(db, client, jwt):
...
+        content_type='application/vnd.api+json',
+        headers=jwt,
+        data=data,
+    )
+
+    db.session.refresh(attendee)
+
+    # Attendee should not be updated
+    assert response.status_code == 422
+    assert attendee.order.id == attendee_order.id
```
Comment (iamareebjamal) [inline tests/all/integration/api/attendee/test_attendee_api.py:594]: Check here that those attributes aren't modified. You copied a test which was testing that order was not modified. That isn't needed here

File: tests/all/integration/api/attendee/test_attendee_api.py [hunk 5]
```diff
@@ -520,3 +520,75 @@ def test_edit_attendee_order(db, client, jwt):
...
```
Comment (iamareebjamal) [inline tests/all/integration/api/attendee/test_attendee_api.py:558]: Same

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/attendees.py  1
         

Clones added
============
- tests/all/integration/api/attendee/test_attendee_api.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5767154&bid=18875279)

**Comment by iamareebjamal** [comment]
Great, now let's just hope dredd works

**Comment by Haider8** [comment]
It will work

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/attendees.py  1
         

Clones added
============
- tests/all/integration/api/attendee/test_attendee_api.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5767154&bid=18875279)

**Comment by iamareebjamal** [comment]
Dredd tests are still failing

**Comment by iamareebjamal** [comment]
Working. Saw old status

**Pull request merged**